src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+import exceptions.DukeDeadlineException;
+import exceptions.DukeException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Constructor
+     * Create a storage file in given filepath
+     *
+     * @param filepath path to storage file
+     * @throws DukeException when does not exist the directory or file
+     */
+    public Storage(String filepath) throws DukeException {
+
+//        String directory_path = "src/main/data";
+//        String directory_path = "./";
+//        File directory = new File(directory_path);
+//        if (!directory.exists()) {
+//            directory.mkdir();
+//            throw new DukeException("No such directory but let me make one for you !");
+//        }
+        File store = new File(filepath);
+        try {
+            store.createNewFile();
+            System.out.println("File is created!");
+        } catch (IOException e) {
+            throw new DukeException("No such storage file! Or maybe file exists already!");
+        }
+
+        this.file = store;
+    }
+
+    /**
+     * Read throw all Task in Storage.file and return an ArrayList of existing Task
+     *
+     * @return ArrayList<Task> all Task in Storage.file
+     * @throws IOException
+     * @throws DukeDeadlineException
+     * @throws DukeException
+     */
+    public ArrayList<Task> getAllTasks() throws IOException, DukeDeadlineException, DukeException {
+        ArrayList<Task> all = new ArrayList<Task>();
```

Comment:
```suggestion
        ArrayList<Task> allTasks = new ArrayList<Task>();
```
all should be plural in this case, as it is a collection of tasks